64.md

```diff
@@ -0,0 +1,85 @@
+NIP-64
+======
+
+Trust Score
+-----------
+
+A trust score is a public way to individually assign a quantifiable level of trust on specifc topics to a nostr identity.
+Ranks can be used to discover trusted people on a common topic.
+
+## Event Structure
+
+The `kind:30382` is a NIP-61 ["Set Item Reference"](https://github.com/arthurfranca/nips/blob/bunch-of-events/61.md#tagging-into-event-sets) event used to assign public information about a pubkey (set to the `d` tag).
+
+This NIP makes use of one or more trust scores added to such events as `T` tags using the following structure:
+
+```js
+  [
+    "T",
+    "<integer_score_ranging_from_-3_to_3><colon_separator><topic_name>",
+    "<+_or_?_or_-><space_separator><review>" // optional and may appear many times
+  ]
+```
+
+Topic names should be preferably in English, lower case and their words are dash separated.
+Topic names are separated from the score by a colon (`:`).
+Some topic names include a context prefix, also separated by a colon (`:`).
+
+After the tag value, there may be any number of reviews, each one with a prefix followed by a space separator
+and the actual review text.
+The prefix denote a postive review (`+`), a negative review (`-`) or a neutral one (`?`).
+
+The `?` prefix is usually assigned for free form user generated reviews. On the other hand, `+` and `-` are usually
+related to clickable fixed options made available by a client.
+
+Example:
+
+```js
+{
+  "kind": 30382,
+  "tags": [
+    ["d", "<pubkey>"],
+    ["T", "3:wiki:role-playing-video-game"], // has written a based wiki article about RPG
+    ["T", "2:car:driver", "+ Skilled", "+ Polite", "- Talks too much", "? Likes to talk about movies."],
+    ["T", "3:buyer", "? No problems when selling to this person"],
+    ["T", "-2:seller", "? Had a bad experience when buying from this person."],
+    ["T", "0:inn-host"], // This person hosted me but I assigned no trust score yet
+    ["T", "3:inn-guest"] // I hosted this person and had no problem
+  ],
+  // ...other fields
+}
+```
+
+If a client doesn't recognize a topic name, it should leave the corresponding `T` tag untouched when editing other entry.
```

Comment:
If a client doesn't recognize a topic name it will probably be discarded when updating the event. We shouldn't rely on devs being altruistic enough to write extra code just for the sake of not nuking content of other clients.
We also shouldn't overload events. A wiki client doesn't care about `car` reviews, so they shouldn't be forced to download it when fetching wiki reviews.

Maybe add the topic to the d tag to separate reviews like `["d", "<top-level-topic>:<pubkey>"]` ?

Reply:
This all is yet to be implemented, feel free to test other ideas derived or not from this.

`["d", "<top-level-topic>:<pubkey>"]` works too. Most pubkeys wouldn't have multiple `T` tags, in fact if using `kind:30382` most would have none. Usually an user won't add that many metadata about another user. That's why currently I think stuffing diverse metadata inside the same kind with `d-tag=another-user` isn't a problem.